### PR TITLE
feat(json-rpc): add json-rpc package

### PR DESCRIPTION
**Pull request by ludamad**
# Description

Initial package
Allows JSON transport of registered types that define toString().

# Checklist:

- [x] I have reviewed my diff in github, line by line.
- [x] Every change is related to the PR description.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to the issue(s) that it resolves.
- [x] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [x] The branch has been rebased against the head of its merge target.
- [x] I'm happy for the PR to be merged at the reviewer's next convenience.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,14 +6,15 @@
   "type": "module",
   "exports": {
     ".": "./dest/index.js",
-    "./eslint": "./.eslintrc.cjs",
-    "./prettier": "./.prettierrc.json"
+    "./eslint": "./estlint-config/index.cjs",
+    "./prettier": "./prettier-config/index.cjs",
+    "./json-rpc": "./json-rpc/src/index.js"
```
Comment (LeilaWang) [inline package.json:11]: Should just be exported from `src/index.ts`?

File: src/json-rpc/client/json_rpc_client.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    {
+      get:
+        (_, rpcMethod: string) =>
+        (...params: any[]) => {
+          debug(`JsonRpcClient.constructor`, 'proxy', rpcMethod, '<-', params);
+          return request(rpcMethod, params);
+        },
+    },
+  ) as RemoteObject<T>;
+}
```
Comment (LeilaWang) [inline src/json-rpc/client/json_rpc_client.ts:87]: What will the type of this Proxy be in a production build? Can it just be `as T`?

File: src/json-rpc/test/test_state.ts
Comment (LeilaWang) [inline src/json-rpc/test/test_state.ts:1]: Rename folder to `fixtures` so that this will be excluded from production build.

**Review by LeilaWang — CHANGES_REQUESTED** [review]

File: src/json-rpc/convert.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  if (!obj) {
+    return obj; // Primitive type
+  }
+  // Is this a serialized Node buffer?
+  if (obj.type === 'Buffer' && typeof obj.data === 'string') {
+    return Buffer.from(obj.data, 'base64');
+  }
+  // Is this a convertible type?
+  if (typeof obj.type === 'string' && typeof obj.data === 'string') {
+    return cc.toClassObj(obj);
```
[the remark this answers is not in the document]
Comment (ludamad) [reply src/json-rpc/convert.ts:20]: Sure, doable

File: package.json (hunk 1 above)
Comment (ludamad) [reply package.json:11]: Your call :) Sure

**Pull request merged**